front/pages/api/w/[wId]/search.ts

```diff
@@ -19,25 +22,192 @@ export type DataSourceContentNode = ContentNodeWithParent & {
   dataSource: DataSourceType;
   dataSourceViews: DataSourceViewType[];
 };
-
 export type PostWorkspaceSearchResponseBody = {
   nodes: DataSourceContentNode[];
   warningCode: SearchWarningCode | null;
   nextPageCursor: string | null;
   resultsCount: number | null;
 };
 
+interface UnifiedSearchStreamChunk {
+  knowledgeResults?: {
+    nodes: DataSourceContentNode[];
+    warningCode: SearchWarningCode | null;
+    nextPageCursor: string | null;
+    resultsCount: number | null;
+  };
+  toolResults?: ToolSearchResult[];
+}
+
+async function handleStreamingSearch(
+  req: NextApiRequest,
+  res: NextApiResponse,
+  auth: Authenticator
+): Promise<void> {
+  const {
+    query,
+    limit: limitParam,
+    cursor,
+    viewType = "all",
+    spaceIds: spaceIdsParam,
+    includeDataSources = "true",
+    searchSourceUrls,
+    includeTools = "true",
+  } = req.query;
+
+  if (typeof query !== "string" || query.length < 3) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Query must be at least 3 characters.",
+      },
+    });
+  }
+
+  const limit = parseInt((limitParam as string) || "25", 10);
+  if (isNaN(limit) || limit < 1 || limit > 100) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "limit must be a number between 1 and 100.",
+      },
+    });
+  }
+
+  try {
+    res.writeHead(200, {
+      "Content-Type": "text/event-stream",
+      "Cache-Control": "no-cache",
+      Connection: "keep-alive",
+    });
+    res.flushHeaders();
+
+    // Create an AbortController to handle client disconnection
+    const controller = new AbortController();
+    const { signal } = controller;
+
+    // Handle client disconnection
+    req.on("close", () => {
+      controller.abort();
+    });
+
+    // Parse spaceIds from query parameter
```

Comment:
NIT: some comments here from our friends could be cleaned up no? Some are really straightforward 🙈